src/libs/PersonalDetailsUtils.js

```diff
@@ -91,4 +95,58 @@ function getLoginsByAccountIDs(accountIDs) {
     );
 }
 
-export {getDisplayNameOrDefault, getPersonalDetailsByIDs, getAccountIDsByLogins, getLoginsByAccountIDs};
+/**
+ * Given a list of logins and accountIDs, return Onyx data for users with no existing personal details stored
+ *
+ * @param {Array<string>} logins Array of user logins
+ * @param {Array<number>} accountIDs Array of user accountIDs
+ * @returns {Object} - Object with optimisticData, successData and failureData (object of personal details objects)
+ */
+function getNewPersonalDetailsOnyxData(logins, accountIDs) {
+    const optimisticData = {};
+    const successData = {};
+    const failureData = {};
+
+    _.each(logins, (login, index) => {
+        const accountID = accountIDs[index];
+
+        if (_.isEmpty(allPersonalDetails[accountID])) {
+            optimisticData[accountID] = {
+                login,
+                accountID,
+                avatar: UserUtils.getDefaultAvatarURL(accountID),
+                displayName: login,
+            };
+
+            // Cleanup the optimistic user to ensure it does not permanently persist
```

Comment:
I think this comment is incomplete. Can you please indicate that this is done to prevent duplicated entries since the server will return other personal details with the correct account ids.